iroh/src/node/builder.rs

```diff
@@ -630,15 +595,25 @@ where
         // ConnectionError::LocallyClosed.  All streams are interrupted, this is not
         // graceful.
         let error_code = Closed::ProviderTerminating;
-        server
+        endpoint
             .close(error_code.into(), error_code.reason())
             .await
             .ok();
+
+        // Shutdown sync and blobs.
+        inner.sync.shutdown().await.ok();
+        inner.db.shutdown().await;
+
+        // Shutdown protocol handlers.
+        protocols.shutdown().await;
+
+        // Abort remaining tasks.
+        join_set.shutdown().await;
```

Comment:
should the above all be raced, or do you want to wait on each serially?

Reply:
I changed it to racing all shutdowns, apart from the join set this comes after, as potentially the shutdown might want to terminate tasks running there more gracefully (the latter is not the case at the moment though).